spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -104,95 +93,135 @@ function AuthenticateTx(msgs []Any, portId string) error {
-3. A relayer listening for `OnChannelOpenInit` events will begin the channel creation handshake
-4. During the `OnChanOpenTry` callback on the host chain an interchain account will be registered and a mapping of the interchain account address to the owner account address will be stored in state (this is used for authenticating transactions on the host chain at execution time)
-5. During the `OnChanOpenAck` & `OnChanOpenConfirm` callbacks on the controller & host chains respectively, the `active-channel` for this interchain account/owner pair, is set in state
+Due to how the mechanics of ICS-004 channel version negotiation operate the version passed into the host chain side (OnChanOpenTry) must match the counterparty version passed into the controller side (OnChanOpenAck) otherwise, there will be a resulting error at the IBC protocol level. 
+
+Combined with the one channel per interchain account approach, this method of version negotiation allows us to pass the address of the interchain account back to the controller chain and create a mapping from controller port ID -> interchain account address during the `OnChanOpenAck` callback. As outlined in the [controlling flow](#Controlling-Flow), a controller chain will need to know the address of a registered interchain account in order to send transactions to the account on the host chain.
+
+#### Version Negotiation Summary
+
+`interchain-account-address` is the address of the interchain account registered on the host chain by the controller chain.
+
+| Initiator | Datagram         | Chain acted upon |  Version (Controller, Host) |
+| --------- | ---------------- | ---------------- |  ---------------------- | 
+| Controller| ChanOpenInit     | Controller       |  (ics27-1, none)        | 
+| Relayer   | ChanOpenTry      | Host             |  (ics27-1, ics27-1.{interchain-account-address})        | 
+| Relayer   | ChanOpenAck      | Controller       |  (ics27-1, ics27-1.{interchain-account-address})        | 
+| Relayer   | ChanOpenConfirm  | Host             |  (ics27-1, ics27-1.{interchain-account-address})        | 
 
+The channel `version` string passed into the `OnChannelOpenTry` callback by a relayer will contain the account address of the interchain account that will be registered on the host chain. A relayer will then pass this version into the `OnChannelOpenAcknowledgement` step on the controller chain as the `counterpartyVersion`. The controller chain will then proceed to parse the interchain account address from the `counterpartyVersion` string and store this address in state with a map to the associated port id.  
 
-**Controlling Flow**
+#### Controlling Flow
 
-Once an interchain account is registered on the host chain a controller chain can begin sending instructions (messages) to control this account. 
+Once an interchain account is registered on the host chain a controller chain can begin sending instructions (messages) to the host chain to control the account. 
```

Comment:
What does _registered_ mean? Does it mean that `ChanOpenConfirm` executed at the host chain?